src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -383,7 +397,24 @@ let rec CheckTypeDeep (cenv: cenv) (visitTy, visitTyconRefOpt, visitAppTyOpt, vi
         CheckTypeDeep cenv f g env typeInstParentOpt body
         tps |> List.iter (fun tp -> tp.Constraints |> List.iter (CheckTypeConstraintDeep cenv f g env))
 
-    | TType_measure _ -> ()
+    | TType_measure tm ->
+        let checkAttribs tm m =
+            match tm with
+            | MeasureAttrib attribs when not attribs.IsEmpty -> CheckFSharpAttributes g attribs m |> CommitOperationResult
```

Comment:
This feels misplaced a little - why is the check not done closer to where Obsolete is checked for all other files?
This function is mainly a recursive visitor, `CheckFSharpAttributes` only for measures (and no other category of TTypes) doesn't sit here well IMO.

Reply:
@T-Gro  The reason for having it here is to raise the warnings/errors in both `expr` and `declarations`. Otherwise we would have to duplicate the checks(CheckExpressions.fs and CheckDeclarations.fs). If that is what you prefer I can do that ?.

Reply:
The downside of moving this check to `CheckExpressions.fs` is that we can not longer raise warnings/error for `m` or `s`. At least I can not find a way to report it.
```fsharp
let genericSumUnits2 (x : float<m>) (y: float<s>) = ()
```

Reply:
I found a way to report this warnings/errors too.

Reply:
This is done now